### ISSUE TITLE
Capacitor google maps plugin not showing map in iphone simulator using Macbook m1

### ISSUE THREAD

**Issue by Nykz** [ionic-team/capacitor-plugins#980]
It shows a blank screen and also when using setCamera() the app crashes...
![IMG-6243](https://user-images.githubusercontent.com/19989930/168730852-5ea209b5-fe56-46c8-8a18-4c944554504a.PNG)
.

My code is available in this github repo
https://github.com/Nykz/ionic-capacitor-googlemaps-android-ios

**Comment by Nykz** [comment]
On real device it’s not showing that’s why I raised the question

**Comment by muuvmuuv** [comment]
Ah, sorry, thought you used an Emulator since you wrote "on MacBook".

**Comment by jongbonga** [comment]
@muuvmuuv 

Try 

```
  ionViewDidEnter() {
    setTimeout(() => {
      this.createMap();
    }, 10);
  }
```

**Comment by jongbonga** [comment]
@Nykz Did you run `npx cap sync` ? I had the same issue until I synched the pod and added the timeout

**Comment by dallastjames** [comment]
Hey @Nykz I took a look at your repo. I believe the issue is caused because you have `capacitor-google-maps` instead of `capacitor-google-map` (plural vs not). Because of this, it wont use the web component and will be missing the structure needed for iOS to create and embed the map. 👍

**Comment by jongbonga** [comment]
![Screenshot 2022-05-18 at 17 42 20](https://user-images.githubusercontent.com/12111716/169085079-582e3975-8057-4f20-b3c9-0aaffd054e44.png)
@Nykz here is your app running fine on my M1.

I changed your HTML to this

```
<ion-content>
  <capacitor-google-maps #map></capacitor-google-maps>
</ion-content>

<ion-footer>
  <ion-toolbar>
    <ion-button expand="block" (click)="locate()">Locate Me</ion-button>
  </ion-toolbar>
</ion-footer>
```

And your TS to

```
  ionViewDidEnter() {
    setTimeout(() => {
      this.createMap();
    }, 1000);
  }
```

make sure to `npx cap sync`

**Comment by Nykz** [comment]
@jongbonga thanks for your response, i tried the yesterday and even i tried just now with all the changes that you showed but not working in my one, can you contact me now via facebook messenger at https://www.facebook.com/profile.php?id=100005738530768 or via any other medium just let me know

**Comment by thomasvidas** [comment]
We'll leave this issue open for visibility, but from Google themselves, [Google Maps doesn't display on iOS simulators on M1 Macs](https://github.com/googlemaps/google-maps-ios-utils/issues/355#issuecomment-1060959728). It will work on a physical devices

**Comment by muuvmuuv** [comment]
I am getting: 

```
CapacitorGoogleMaps/Map.swift:274: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value
2022-05-21 11:18:53.536753+0200 App[4828:321965] CapacitorGoogleMaps/Map.swift:274: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value
(lldb) 
```

It seems that `GMapView` is undefined here, it does not get initiated. I am on my iPhone, no Simulator.

**Comment by muuvmuuv** [comment]
Ok, it seems I just had to do a setTimeout like suggested. @thomasvidas is there any "onReady" event we can listen to until the web component is ready? On slow devices I would need to increase the setTimeout and on fast decrease...

**Comment by Nykz** [comment]
@jongbonga its not working in my mac M1 after applying all your changes too, its still the same, there is a serious issue that's going on not only with me but with other people too. So kindly if there is any other solution do let me know...Gmapview not getting rendered in uiview, also current location functionality too not asking for permission even after making changes in info.plist

**Comment by muuvmuuv** [comment]
My problem now is that the map just disappears when using tab navigation :D

**Comment by timeisgolden** [comment]
@muuvmuuv I am facing the same issue as you.
the map is disappearing when changing tab navigation. 
Did you solve the issue?

**Comment by muuvmuuv** [comment]
@timeisgolden yes, I did. I have created a new render cycle by re-creating the map when the user comes back into the view with `ionWillEnter` and `ionDidLeave`.

Still, this method is inefficient because it forces the map to re-create all markers and tiles, which will also cause a flicker of the map and some time until it's loaded. Additionally, I have CSS set up that shows a loading indicator beneath the map, so it gets hidden when it finishes.

This will be somewhere in your map component. In your feature component, you would like to use `ionViewDidEnter` to call below again and again. And `ionViewWillLeave` to always make sure it gets destroyed on leaving the view.

```ts
    // BUG: Wait for UIViewController to be ready - https://github.com/ionic-team/capacitor-plugins/issues/980
    await sleep(0.345)

    if (this.#map) {
      await this.#map.removeAllMapListeners()
      await this.#map.destroy()
      this.#map = undefined
    }

    this.log.info(StackGlobal.Map, `Create map ${this.name}`)

    try {
      this.#map = await GoogleMap.create({
        id: this.name,
        element: this.mapRef.nativeElement,
        apiKey: environment.googleApiKey,
        forceCreate: true,
        config: {
          center: { lat, lng },
          zoom: this.zoom,
          styles: mapStyles,
        },
      })
    } catch (error) {
      this.log.error(StackGlobal.Map, 'Could not create map', error)
      return
    }
```

**Comment by jcesarmobile** [comment]
google maps works on simulators now, for the other issues please report them on https://github.com/ionic-team/capacitor-google-maps/issues if they are still relevant, but please, provide a sample app that reproduces the issue
